cmd/releasego/publish-announcement.go

```diff
@@ -200,23 +198,46 @@ func publishAnnouncement(p subcmd.ParseFunc) (err error) {
 		return fmt.Errorf("file %s already exists in go-devblog repository", blogFilePath)
 	}
 
-	if err := githubutil.Retry(func() error {
-		// Upload the announcement to the go-devblog repositoryy main branch with proper commit message
-		if err := githubutil.UploadFile(
-			ctx,
-			client,
-			org,
-			repo,
-			"main",
-			blogFilePath,
-			fmt.Sprintf("Add new blog post for new release in %s", releaseDate.Format("2006-01-02")),
-			content.Bytes()); err != nil {
-			return fmt.Errorf("error uploading file to go-devblog repository : %w", err)
-		}
+	// Create a feature branch (gitpr convention: dev/<purpose>/<name>) and open a PR to main.
+	prSet := gitpr.PRRefSet{Name: "main", Purpose: fmt.Sprintf("blog-%d", time.Now().Unix())}
```

Comment:
I took a look at the code, and actually you're right, the comment on the API is correct, `Name: "main"` is right because it is used as the base. Just a weird key name. However, `Purpose` should contain the slug, for clarity when browsing branches (e.g. to clean up). I think `"blog/%v/%d"` with the slug and timestamp would make the most sense to me.

(I prefer `/` over `-` in that format string because it works as a directory separator in some contexts, which helps navigation.)